### PR TITLE
Clarify Live Debugger and Dynamic Instrumentation docs

### PR DESCRIPTION
**Pull request by timsee**
### What does this PR do? What is the motivation?

We've had a lot of product decisions around live debugger and dynamic instrumentation, but havent formalized them in external docs. Changes in here are mainly around:
- what is live debugger vs dynamic instrumentation, and removing cases where we use DI to refer to both
- specifying that LD is not just a UI feature, it also has an agent and a mcp you can use
- redaction modes and how they are handled
- some edge cases around the expression language 

Five English documentation pages only. The scrubbing page moves under Live Debugger with redirects from its old URLs. Simulator code and test changes are deferred.

### Merge readiness

- [ ] Ready for merge

### AI assistance

AI-assisted drafting, implementation checks, and automated validation.

### Additional notes

- Markdoc, the affected-page Hugo build, 101 existing JavaScript tests, and 70 rendered-page and redirect checks passed. No new Vale alerts on changed lines.
- Checked 18 expression examples and the corrected Python startup command using ddtrace 4.11.0. Other runtime suites and live MCP enablement were not rerun for this prose revision.
- Complements #40033 without duplicating its MCP tool catalog.

### CODE REVIEW

File: hugo/content/en/tracing/trace_collection/dynamic_instrumentation/expression-language.md [hunk 1]
```diff
@@ -80,21 +86,23 @@ The following examples assume a variable named `mySequence` with value `[1,2,3,4
...
+Expression support and missing-value behavior can differ in your application.
+
 Select one of the examples or enter an expression in the "when" field and click "SIMULATE" to see whether data is captured based on your condition.
 
 Available variables in this example:
 
 - `loops`: The route parameter hardcoded to `5`
 - `myString`: A string `"Hello, world!"`
-- `mySequence`: An array of integers `[1, 2, 3]`
+- `mySequence`: An array of integers `[1, 2, 3, 4]`
```
Comment (timsee) [inline hugo/content/en/tracing/trace_collection/dynamic_instrumentation/expression-language.md:105]: such a small nit, but the agents are annoyed: we reference `mySequence[3]` but then give a len == 3 sequence so thats an out of bounds.

File: hugo/content/en/tracing/trace_collection/dynamic_instrumentation/expression-language.md [hunk 2]
```diff
@@ -34,25 +39,26 @@ Expressions can be used to produce metrics and as conditions to emit filtered da
...
+The available contextual variables depend on the instrumentation location. `@return` and `@duration` are available at method exit. `@it` is used inside collection predicates.
 
 | Keyword     | Description                                                                |
 |-------------|----------------------------------------------------------------------------|
 | `@return`   | Provides access to the method return value. |
 | `@duration` | Provides access to the method call execution duration, as a floating-point value in milliseconds. |
 | `@exception`| Provides access to the exception thrown within the method (only available if an uncaught exception exists). |
 | `@it`       | Provides access to the current element during collection iteration. Used in predicates for list operations. |
-| `@key`      | Provides access to the current key during dictionary iteration. Used in predicates for dictionary operations. |
```
Comment (timsee) [inline hugo/content/en/tracing/trace_collection/dynamic_instrumentation/expression-language.md:54]: this was never fully completed, so we currently reject them in all [maybe most?] cases

File: hugo/content/en/tracing/live_debugger/_index.mdoc.md [hunk 3]
```diff
@@ -372,7 +378,13 @@ Logpoints are "non-breaking breakpoints" that specify where in the code to captu
 
 Most logpoint settings can be modified after creation, even if the logpoint already started capturing log events. However, the logpoint's originally selected service, environment, and code location cannot be modified (a new logpoint or Debug Session should be created in this case).
 
-After a logpoint is created, modified, or re-activated, it can take a couple of minutes to instrument the code and begin capturing log events. **Note**: If the selected code is not executed or the logpoint condition(s) are not met, then no log events are generated.
+After a logpoint is created, modified, or re-activated, it can take a couple of minutes to instrument the code and begin capturing log events. If no useful data appears, check that:
+
+- The selected service and environment are receiving traffic that executes the instrumented code.
```
Comment (timsee) [inline hugo/content/en/tracing/live_debugger/_index.mdoc.md:383]: is this too much? i kind of think it is... but also think nothing is too little?

File: hugo/content/en/tracing/live_debugger/sensitive-data-scrubbing.md [hunk 4]
```diff
@@ -1,31 +1,33 @@
 ---
-title: Dynamic Instrumentation Sensitive Data Scrubbing
-description: Protect sensitive information by configuring redaction and scrubbing mechanisms for Dynamic Instrumentation.
+title: Live Debugger Sensitive Data Scrubbing
```
Comment (tylfin) [inline hugo/content/en/tracing/live_debugger/sensitive-data-scrubbing.md:2]: This file path probably needs to be updated

File: hugo/content/en/tracing/live_debugger/_index.mdoc.md [hunk 5]
```diff
@@ -49,13 +49,15 @@ Instead of adding temporary debug logs or reproducing issues locally, you can dy
 
 Live Debugger uses logpoints: auto-expiring, non-breaking breakpoints that collect diagnostic data without pausing the application. Since execution continues normally, Live Debugger can be used safely on production systems to investigate problems as they happen.
 
+Live Debugger complements [Dynamic Instrumentation][38], which adds metrics, spans, and span tags using the same [expression language][15].
```
Comment (sstonehill12) [inline hugo/content/en/tracing/live_debugger/_index.mdoc.md]: @timsee To improve the agent's understanding of DI vs. LD, does the explanation need to be in the intro section? Or could we have a dedicated section lower on the page with a more detailed explanation of how they overlap and differ?

Comment (timsee) [reply hugo/content/en/tracing/live_debugger/_index.mdoc.md]: yeah, lets move out of the intro do a dedicated section, thats a great idea!

File: hugo/content/en/tracing/live_debugger/_index.mdoc.md [hunk 6]
```diff
@@ -49,13 +49,15 @@ Instead of adding temporary debug logs or reproducing issues locally, you can dy
...
+
 ## Key capabilities
 
 Live Debugger provides:
 
 - **Real-time inspection** of variable values, method arguments, and execution context in running code.
 - **Safe, non-invasive data capture** that collects debugging information without pausing applications or requiring redeploys.
-- **Dynamic logpoint placement** anywhere in your codebase, including in third-party libraries.
+- **Dynamic logpoint placement** at supported code locations in your application or third-party libraries.
```
Comment (sstonehill12) [inline hugo/content/en/tracing/live_debugger/_index.mdoc.md]: I would move this up to the first bullet so that the following bullets read as a more detailed explanation of what these logpoints do. Plus the copy changes below.

```suggestion
- **Dynamic logpoints** (non-breaking breakpoints) placed at any supported code location you choose in your application or third-party libraries 
```

File: hugo/content/en/tracing/live_debugger/_index.mdoc.md (hunk 3 above)
Comment (sstonehill12) [reply hugo/content/en/tracing/live_debugger/_index.mdoc.md]: I think details like this are helpful, especially for agent context. I might want to move this to a Troubleshooting section though rather than lumping it into the "Creating logpoints" section. What do you think?

File: hugo/content/en/tracing/live_debugger/bits-live-debugger.md [hunk 7]
```diff
@@ -39,7 +39,8 @@ Bits Live Debugger uses <a href="/bits_ai/bits_code/">Bits Code</a> as the under
 Before using Bits Live Debugger:
 
 - [Live Debugger][1] must be enabled for the target service. See [Enable Live Debugger][7] for details.
-- Your account must have the [permissions][2] required to use Live Debugger, including read, write, and variable-capture permissions for the target environment.
+- Your account must have the [permissions][2] required to use Live Debugger in the target environment.
```
Comment (sstonehill12) [inline hugo/content/en/tracing/live_debugger/bits-live-debugger.md]: This "target environment" language is probably referring to the debugger_write_preprod permission.

```suggestion
- Your account must have the [permissions][2] required to use Live Debugger.
```

File: hugo/content/en/tracing/live_debugger/bits-live-debugger.md [hunk 8]
```diff
@@ -39,7 +39,8 @@ Bits Live Debugger uses <a href="/bits_ai/bits_code/">Bits Code</a> as the under
...
+- Agentic investigations require [Targeted redaction mode][8]. Manual Live Debugger sessions work in both Strict and Targeted Mode.
```
Comment (sstonehill12) [inline hugo/content/en/tracing/live_debugger/bits-live-debugger.md]:
```suggestion
- The selected service and environment for the Bits debug session are set to [Targeted redaction mode][8]. Manually created Live Debugger sessions work in both Strict and Targeted redaction modes.
```

File: hugo/content/en/tracing/live_debugger/bits-live-debugger.md [hunk 9]
```diff
@@ -78,16 +79,17 @@ Logpoints created by Bits follow the same rules as manually created logpoints. T
...
 
 [1]: /tracing/live_debugger/
 [2]: /tracing/live_debugger/#permissions
-[3]: /dynamic_instrumentation/sensitive-data-scrubbing/
+[3]: /tracing/live_debugger/sensitive-data-scrubbing/
 [4]: https://app.datadoghq.com/debugging/
 [5]: /bits_ai/bits_code/
 [6]: /source_code/source-code-management/
 [7]: /tracing/live_debugger/#enable-live-debugger
+[8]: /tracing/live_debugger/#mode-based-redaction
```
Comment (sstonehill12) [inline hugo/content/en/tracing/live_debugger/bits-live-debugger.md]:
```suggestion
[8]: /tracing/live_debugger/#mode-based-redaction
[9]: /tracing/live_debugger/#protecting-sensitive-data
```

File: hugo/content/en/tracing/live_debugger/bits-live-debugger.md [hunk 10]
```diff
@@ -78,16 +79,17 @@ Logpoints created by Bits follow the same rules as manually created logpoints. T
 
 **Language support**: Some features vary by language. For example, condition expressions are not supported for all runtimes. Bits notifies you when a requested feature is not available for the target service's language.
 
-**Sensitive data**: The [sensitive data scrubbing][3] behavior that applies to manually created logpoints also applies to logpoints created by Bits. In production environments, non-numeric and non-Boolean captured values are redacted by default.
+**Sensitive data**: Logpoints created by Bits use [sensitive data scrubbing][3]. In production environments, non-numeric and non-Boolean captured values are redacted by default. Targeted Mode still redacts sensitive data. It does not disable redaction. Changing the mode requires the **Live Debugger Redaction Write** permission.
```
Comment (sstonehill12) [inline hugo/content/en/tracing/live_debugger/bits-live-debugger.md]: This is misleading because Bits won't create logpoints for service/envs in Strict mode, and the "production environments" sentence describes Strict mode. Also I think its better to direct the user to the LD doc "Protecting Sensitive Data" section, which then links to the sensitive-data-scrubbing.md page.

```suggestion
**Sensitive data**: Bits will only initiate a debug session on a service and environment with Targeted redaction mode (not Strict mode). The logs generated by Bits debug sessions will apply Targeted redaction, which still [redacts sensitive data][9]. Changing a redaction mode requires the **Live Debugger Redaction Write** permission.
```

File: hugo/content/en/tracing/live_debugger/sensitive-data-scrubbing.md
Comment (sstonehill12) [inline hugo/content/en/tracing/live_debugger/sensitive-data-scrubbing.md:1]: It might make sense to move all of the "Protecting Sensitive Data" content from the parent LD doc into this child doc.

File: hugo/content/en/tracing/trace_collection/dynamic_instrumentation/_index.mdoc.md [hunk 11]
```diff
@@ -49,6 +49,8 @@ further_reading:
 
 Dynamic Instrumentation lets you add metrics, spans, and span tags to running production systems without restarts or code changes, including in third-party libraries.
 
+Dynamic Instrumentation complements [Live Debugger][23], which captures logs and variable snapshots using the same [expression language][15].
```
Comment (sstonehill12) [inline hugo/content/en/tracing/trace_collection/dynamic_instrumentation/_index.mdoc.md]: Same comment as on the LD doc - we can duplicate the same copy on both pages if the docs team is okay with it.

File: hugo/content/en/tracing/live_debugger/_index.mdoc.md (hunk 5 above)
Comment (sstonehill12) [reply hugo/content/en/tracing/live_debugger/_index.mdoc.md]: Working on a draft for this, will add a comment here when ready.

Comment (sstonehill12) [reply hugo/content/en/tracing/live_debugger/_index.mdoc.md]: Let me know what you think for a separate section @timsee -


## Live Debugger vs. Dynamic Instrumentation

Live Debugger and Dynamic Instrumentation are built on the same instrumentation technology and share some overlapping configuration settings and conventions, such the same [expression language][1] for log message templates and conditions. Both collect data from a code location you choose in a running service, with no code changes, redeploys, or restarts.

They differ in what they capture and how long it lasts. Use Live Debugger to investigate a problem: logpoints capture log events and variable snapshots, and expire when the Debug Session ends. Use Dynamic Instrumentation to add APM custom instrumentation without modifying source code: dynamic spans, span tags, and metrics stay active until you disable them.

Comment (timsee) [reply hugo/content/en/tracing/live_debugger/_index.mdoc.md]: I love it! Will probably backport some of this phrasing to our skills as well, since its much less clear in those. 

The only point I'm unclear on is treating Debug Session as a proper noun. is... it? but copying this in with that switched to lowercase (unless we have a product called Debug Session I'm unaware of... datadog does like to name products really simple things 😂 ) . feel free to push back if thats intentional!

Comment (sstonehill12) [reply hugo/content/en/tracing/live_debugger/_index.mdoc.md]: ~Lowercase "debug session" is fine with me~

Ah actually let's ask the docs team to confirm when they do their review. It's a nit picky thing but I'm seeing "Debug Session" in a lot of places in the LD docs so maybe we stick with that convention..

**Comment by git-thuerk-done** [comment]
Editorial review tracked in [DOCS-15787](https://datadoghq.atlassian.net/browse/DOCS-15787).

[DOCS-15787]: https://datadoghq.atlassian.net/browse/DOCS-15787?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
